src/pages/posts/[slug].js

```diff
@@ -98,10 +120,21 @@ export async function getStaticProps({ params = {} } = {}) {
 
   const socialImage = `/images/${params?.slug}.png`;
 
+  const { categories, postId } = post;
+  const category = categories.length && categories[0];
+  const { name, slug } = category;
+
   return {
     props: {
       post,
       socialImage,
+      relatedPosts: {
+        posts: await getRelatedPosts(category, postId),
+        title: {
+          name: name || '',
```

Comment:
maybe we leave this as `undefined` and only show `More Posts` if it's not defined? rather than it showing `More from ` where it would include a link with no text in it, if im understanding this correctly

maybe if it's undefined `More Posts` is the whole link

Reply:
I see your point, I can change that and set the title with a conditional render evaluating if there is a `name`. Just two considerations about that:
- When rendering the `relatedPosts`, first check is if the array of post are not empty. And I was thinking if is possible to have a case in which there is an array of posts but the name is `undefined`.  This would mean that the category exists (because the relatedPosts are collected from it), but the name is missing ...  is this a possibility? 

-  I just discovered while doing this that when returning the props at `getStaticProps` it can't have an **undefined** value, because it cannot be serialized. So one option is to pass it like `name: name || null,`

![undefined](https://user-images.githubusercontent.com/50624358/107075825-6a091680-67c9-11eb-92fc-453e5574df49.png)

Reply:
- I'm not sure it's possible to have an undefined category name, but i was thinking if someone hard coded it in there if they wanted to update the title but keep the same link 🤷 
- interesting about undefined, TIL. i think the `name || null` makes sense